From dddee33beee02e4b71aa1a3e6c7df43c7f6376a8 Mon Sep 17 00:00:00 2001
From: mihirlad55 <mihirlad55@gmail.com>
Date: Mon, 24 Aug 2020 02:37:40 +0000
Subject: [PATCH] Update from v1.5.4 to v1.5.5

- Fixed dwm-msg not accepting negative float values
- Removed spawn command from config.def.h since it is currently unusuable and
  does not provide any useful functionality
---
 config.def.h | 1 -
 dwm-msg.c    | 5 +++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 3ad9785..059a831 100644
--- a/config.def.h
+++ b/config.def.h
@@ -123,7 +123,6 @@ static IPCCommand ipccommands[] = {
   IPCCOMMAND(  focusmon,            1,      {ARG_TYPE_SINT}   ),
   IPCCOMMAND(  focusstack,          1,      {ARG_TYPE_SINT}   ),
   IPCCOMMAND(  zoom,                1,      {ARG_TYPE_NONE}   ),
-  IPCCOMMAND(  spawn,               1,      {ARG_TYPE_PTR}    ),
   IPCCOMMAND(  incnmaster,          1,      {ARG_TYPE_SINT}   ),
   IPCCOMMAND(  killclient,          1,      {ARG_TYPE_SINT}   ),
   IPCCOMMAND(  togglefloating,      1,      {ARG_TYPE_NONE}   ),
diff --git a/dwm-msg.c b/dwm-msg.c
index 606f6ce..0071781 100644
--- a/dwm-msg.c
+++ b/dwm-msg.c
@@ -225,14 +225,19 @@ is_float(const char *s)
 {
   size_t len = strlen(s);
   int is_dot_used = 0;
+  int is_minus_used = 0;
 
   // Floats can only have one decimal point in between or digits
+  // Optionally, floats can also be below zero (negative)
   for (int i = 0; i < len; i++) {
     if (isdigit(s[i]))
       continue;
     else if (!is_dot_used && s[i] == '.' && i != 0 && i != len - 1) {
       is_dot_used = 1;
       continue;
+    } else if (!is_minus_used && s[i] == '-' && i == 0) {
+      is_minus_used = 1;
+      continue;
     } else
       return 0;
   }
-- 
2.28.0

